src/HOC/RouteHOC.tsx

```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { Redirect, Route } from "react-router-dom";
+
+const RouteHOC = ({ path, token, redirect, children }: any) => {
```

Comment:
Not sure whether this really a `HOC`, as far as I know:
`a higher-order component is a function that takes a component and returns a new component.`
This seems to be regular component